File: program (104).java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  Function Name  :    Difference
//  Description :       Return Difference between summation of even elements and odd elements. 
//  Input :             int 
//  Output :            int
//  Author :            Aman Shaikh    
//  Date:               14/12/2024 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class ArrayDemo
{
    public int Difference(int Arr[], int iNo)
    {
        int i = 0, iEven = 0, iOdd = 0;
        for(i = 0; i < iNo; i++)
        {
            if(Arr[i]%2 == 0)
            {
                iEven = iEven + Arr[i];
            }
            else if(Arr[i]%2 != 0)
            {
                iOdd = iOdd + Arr[i];
            }
        }
        return iEven-iOdd;
        
    }
}

class Assignment37A
{
    public static void main(String Arr[])
    {
        int iValue = 0, iCnt = 0;
        
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter frequency : ");
        iValue = sobj.nextInt(); 

        int Brr[] = new int[iValue]; 
        System.out.println("Enter the numbers : ");
        for(iCnt = 0; iCnt < iValue; iCnt++)
        {
            Brr[iCnt] = sobj.nextInt(); 
        }

        ArrayDemo aobj = new ArrayDemo();
        int iResult = aobj.Difference(Brr,iValue); 

        System.out.println("Difference is : "+iResult);
    }
}

File: program (104).cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  Function Name  :    WhiteSpace                                        
//  Description :       Accept string from user and count no. of white spaces using Recursion.
//  Input :             string 
//  Output :            int
//  Author :            Aman Shaikh    
//  Date:               15/12/2024 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

class Pattern
{
    public:
    int WhiteSpace(char *str) 
    {
        static int iNum = 0;

        if(*str != '\0') 
        {
            if(*str == ' ') 
            {
                iNum++;
            }
            str++;
            WhiteSpace(str); 
        } 
        return iNum;
    }
};

int main()
{
    int iRet = 0;
    char Arr[30];

    cout<<"Enter the String : "; 
    cin.getline(Arr,30);

    Pattern pobj;
    iRet = pobj.WhiteSpace(Arr);

    cout<<"Count is : "<<iRet<<"\n";

    return 0;
}
